CMakeLists.txt

```diff
@@ -335,6 +335,13 @@ if (OPENSSL_FOUND)
 endif(OPENSSL_FOUND)
 
 add_subdirectory(lib)
+
+# Keep this after lib because lib is made up of third party libraries, so if
+# there are warnings, we can't do anything about it.
+add_compile_options(-pipe -Wall -Wextra -Werror -Wno-unused-parameter -Wno-format-truncation -Wno-cast-function-type -Wno-stringop-overflow)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-noexcept-type>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wsuggest-override>)
```

Comment:
I could not figure out how to add two options to a generator like this without calling add_compile_options twice. I tried a variety of quoting permutations, but nothing worked. If someone else can figure it out, I'll pull it in.

Reply:
```cmake
add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:-Wno-except-type;-Wsuggest-override>"
```